server-spi-private/src/main/java/org/keycloak/authentication/RequiredActionFactory.java

```diff
@@ -44,4 +47,24 @@ public interface RequiredActionFactory extends ProviderFactory<RequiredActionPro
     default boolean isOneTimeAction() {
         return false;
     }
+
+    /**
+     * Indicates whether this required action can be configured via the admin ui.
+     * @return
+     */
+    default boolean isConfigurable() {
+        List<ProviderConfigProperty> configMetadata = getConfigMetadata();
+        if (configMetadata == null) {
+            return false;
+        }
+        return !configMetadata.isEmpty();
+    }
+
+    /**
+     * Allows users to validate the provided configuration for this required action. Users can throw a {@link org.keycloak.models.ModelValidationException} to indicate that the configuration is invalid.
+     * @param realm
+     * @param model
+     */
+    default void validateConfig(RealmModel realm, RequiredActionConfigModel model) {
```

Comment:
I'm not sure whether this is the proper way to perform validation, but it could be used to prevent bad configurations to be written to the database.